app/server.go

```diff
@@ -261,6 +261,22 @@ func NewServer(options ...Option) (*Server, error) {
 		s.StartElasticsearch()
 	}
 
+	s.AddConfigListener(func(oldConfig *model.Config, newConfig *model.Config) {
+		if *oldConfig.GuestAccountsSettings.Enable && !*newConfig.GuestAccountsSettings.Enable {
+			newApp := New(s.AppOptions()...)
```

Comment:
If we don't have time to fix it properly we should use FakeApp instead.

Reply:
Ok, back to FakeApp then